### PR TITLE
net/pkt: Allocate relevant size on AF_UNSPEC

### LINKED ISSUE #12982: net: Zephyr drops IPv4 packet with extended MAC frame size
If Zephyr receives a packet with IPv4 datagram occupying only portion of the MAC frame it drops the packet without responding to it. This is a regression from PR #11775.

As an example, attached also pcap from Linux, which responds correctly.

[ext-frame-icmp-zephyr.pcap.gz](https://github.com/zephyrproject-rtos/zephyr/files/2822242/ext-frame-icmp-zephyr.pcap.gz)
[ext-frame-icmp-linux.pcap.gz](https://github.com/zephyrproject-rtos/zephyr/files/2822243/ext-frame-icmp-linux.pcap.gz)

### PR DESCRIPTION
**Pull request by tbursztyka**
In case of Ethernet, if the size is > than MTU and if AF_UNSPEC is
provided to the allocator will need to take into account the ethernet
header size which is not accounted in the MTU.

Other current L2 do not follow that rule as their MTU is based on IP one
(IPv6 most of the time).

Fixes #12982

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

### CODE REVIEW

**Comment by pfalcon** [comment]
Can the commit message by polished please?

> In case of Ethernet, if the size is

Size of what?

> In case of Ethernet, if the size is >

Can a word instead of ">" be used there?

> and if AF_UNSPEC is
provided to the allocator will need to take into account

This one is really hard to parse. Is allocator is contracted to both an object and subject here? Should it be "and if AF_UNSPEC is
provided to the allocator, *the allocator* will need to take into account" ?

**Comment by pfalcon** [comment]
Beyond that, the change code seems ok, but reasoning behind it is unclear - why is Ethernet so special, that it needs workarounds like that?

**Comment by tbursztyka** [comment]
I'll clarify why only ethernet has this special treatment in the commit msg

**Review by rmstoi — APPROVED** [review]

**Review by pfalcon — APPROVED** [review]
Thanks for much more detailed and clear commit message!

**Review by jukkar — APPROVED** [review]
